mock/mock.py

```diff
@@ -2059,9 +2059,15 @@ def _format_call_signature(name, args, kwargs):
     message = '%s(%%s)' % name
     formatted_args = ''
     args_string = ', '.join([repr(arg) for arg in args])
-    kwargs_string = ', '.join([
-        '%s=%r' % (key, value) for key, value in sorted(kwargs.items())
-    ])
+
+    kwargs_to_join = []
+    for key, value in sorted(kwargs.items()):
+        if six.PY2 and isinstance(key, unicode):
+            key = key.encode("utf8")
```

Comment:
No, because we're doing ``%r`` formatting and not ``%s``. The python documentation states that ["the return value must be a string object"](https://docs.python.org/2/reference/datamodel.html#object.__repr__) so if that's not the case, it's probably ok to crash.